core/src/main/java/org/mskcc/cbio/oncokb/bo/impl/EvidenceBoImpl.java

```diff
@@ -95,18 +99,17 @@ public List<Evidence> findEvidencesByAlteration(Collection<Alteration> alteratio
             relevantMatchedTumorTypes = tumorTypes;
         }
 
-        List<Evidence> sortedEvidences = new ArrayList<>();
+        Set<Evidence> sortedEvidences = new LinkedHashSet<>();
         for (TumorType tumorType : relevantMatchedTumorTypes) {
             for (int i = 0; i < evidences.size(); i++) {
                 Evidence evidence = evidences.get(i);
                 if (evidence.getCancerTypes().contains(tumorType)) {
                     sortedEvidences.add(evidence);
-                    evidences.remove(evidence);
```

Comment:
So basically, this line is messing the index in the loop

Reply:
yup